src/foam/core/types.js

```diff
@@ -491,7 +491,12 @@ foam.CLASS({
     {
       class: 'String',
       name: 'targetDAOKey',
-      expression: function(of) { return foam.String.daoize(of.name); }
+      expression: function(of) {
+        if ( ! of ) {
+          console.log('ERROR: invalid \'of\' for property with targetDAOKey', this.name);
```

Comment:
Wouldn't console.error() be better? If you use " instead of ' then you won't need to escape the 's.